src/sumologic-context-manager/events.ts

```diff
@@ -0,0 +1,192 @@
+      self.IDBDatabase?.prototype,
+      self.IDBTransaction?.prototype,
+      self.IDBCursor?.prototype,
+    ],
+    [
+      'upgradeneeded',
+      'complete',
+      'abort',
+      'success',
+      'error',
+      'blocked',
+      'versionchange',
+      'close',
+    ],
+  ],
+]);
+
+type OnListener = (this: unknown, event: Event) => any;
+
+export const patchEvents = (contextManager: ContextManager) => {
+  const wrappedOnListeners = new WeakMap<OnListener, OnListener>();
+
+  const patchOnProperty = <O extends Object>(
+    object: O,
+    property: string,
+  ): void => {
+    const descriptor = Object.getOwnPropertyDescriptor(object, property);
+    if (!descriptor) {
+      const proto = Object.getPrototypeOf(object);
+      if (!proto) return;
+      return patchOnProperty(proto, property);
+    }
+    wrapWithToString(
+      descriptor,
+      'get',
+      (original) =>
+        function (this: unknown) {
+          const listener = original!.call(this);
+          return wrappedOnListeners.get(listener) || listener;
+        },
+    );
+    wrapWithToString(
+      descriptor,
+      'set',
+      (original) =>
+        function (this: unknown, listener: OnListener | null) {
+          let wrappedListener = listener;
+          if (listener) {
+            wrappedListener = contextManager.bind(
+              contextManager.active(),
+              listener,
+            );
+            wrappedOnListeners.set(wrappedListener, listener);
+          }
+          original!.call(this, wrappedListener);
+        },
+    );
+    Object.defineProperty(object, property, descriptor);
+  };
+
+  const patchOnProperties = <O extends Object>(
+    object: O,
+    properties: string[],
+  ): void => {
+    properties.forEach((property) => {
+      patchOnProperty(object, `on${property}`);
+    });
+  };
+
+  const wrappedEventListeners = new WeakMap<
+    EventListenerOrEventListenerObject,
+    EventListenerOrEventListenerObject
+  >();
+
+  wrapWithToString(
+    EventTarget.prototype,
+    'addEventListener',
+    (original) =>
+      function (this: EventTarget, ...args) {
+        const listener = args[1];
+        if (listener) {
+          let wrappedListener = wrappedEventListeners.get(listener);
+          if (!wrappedListener) {
+            if (typeof listener === 'function') {
+              wrappedListener = contextManager.bind(
+                contextManager.active(),
+                listener,
+              );
+            } else if (listener && typeof listener.handleEvent === 'function') {
+              wrappedListener = {
+                ...listener,
+                handleEvent: contextManager.bind(
+                  contextManager.active(),
+                  listener.handleEvent,
+                ),
+              };
+            }
+          }
+          if (wrappedListener) {
+            wrappedEventListeners.set(listener, wrappedListener);
+            args[1] = wrappedListener;
```

Comment:
I want to mutate the `args` because:
- we don't create new objects,
- it may support unexpected arguments (when some method will be extended in a feature).